src/noLifeCycleCallRule.ts

```diff
@@ -49,19 +50,35 @@ export class ExpressionCallMetadataWalker extends NgWalker {
   }
 
   private validateCallExpression(node: ts.CallExpression): void {
-    const name = (node.expression as any).name;
+    const name = ts.isPropertyAccessExpression(node.expression) ? node.expression.name : undefined;
+    const expression = ts.isPropertyAccessExpression(node.expression) ? node.expression.expression : undefined;
 
-    if (!name || !lifecycleHooksMethods.has(name.text)) {
+    const isSuperCall = expression && expression.kind === ts.SyntaxKind.SuperKeyword;
+    const isNotLifecycleCall = !name || !ts.isIdentifier(name) || !lifecycleHooksMethods.has(name.text as any);
+
+    if (isSuperCall || isNotLifecycleCall) {
       return;
     }
 
-    let currentNode = node as any;
+    let failureString = Rule.FAILURE_STRING;
+
```

Comment:
nit: extra new line.

Reply:
I intentionally separated the if block with blank lines for readability.

Reply:
I don't mark the `if` block, but the block between `let` and `const` variables.

Reply:
I can remove if it violates the project's code conventions.

Reply:
Nah, it's just a nit. :)